src/screens/WalletType.js

```diff
@@ -0,0 +1,216 @@
+ * Screen used to select between hardware wallet or software wallet
+ *
+ * @memberof Screens
+ */
+class WalletType extends React.Component {
+  // Trial parameters when trying to connect to ledger
+  // trialNumber: current trial
+  // trialLimit: maximum number of trials
+  // trialInterval: wait interval between trials
+  trialNumber = 0
+  trialLimit = 100
+  trialInterval = 3000
+
+  /**
+   * errorMessage {string} message to show in case of error connecting to ledger
+   * hardware {boolean} if hardware wallet was chosen
+   * waitAction {boolean} after connecting to ledger we need to wait for user action
+   */
+  state = {
+    errorMessage: '',
+    hardware: false,
+    waitAction: false,
+  }
+
+  componentDidMount() {
+    // Update Sentry when user started wallet now
+    wallet.updateSentryState();
+
+    if (IPC_RENDERER) {
+      IPC_RENDERER.on("ledger:version", (event, arg) => {
+        if (arg.success) {
+          // We wait 2 seconds to update the message on the screen
+          // so the user don't see the screen updating fast
+          setTimeout(() => {
+            this.setState({ waitAction: true, errorMessage: '' }, () => {
+              ledger.getPublicKeyData();
+            })
+          }, 2000);
+        } else {
+          if (!this.state.hardware) {
+            // User clicked 'Back' button
+            this.trialNumber = 0;
+            return;
+          }
+
+          if (this.trialNumber < this.trialLimit) {
+            setTimeout(() => {
+              this.trialNumber += 1;
+              ledger.getVersion();
+            }, this.trialInterval);
+          } else {
+            // Error
+            this.setState({ errorMessage: arg.error });
+          }
+        }
+      });
+
+      IPC_RENDERER.on("ledger:publicKeyData", (event, arg) => {
+        if (!this.state.hardware) {
+          // User clicked 'Back' button
+          return;
+        }
+
+        if (arg.success) {
+          const data = arg.response;
+          const uncompressedPubkey = data.slice(0, 65);
+          const compressedPubkey = hathorLib.wallet.toPubkeyCompressed(uncompressedPubkey);
+          const chainCode = Buffer.from(data.slice(65, 97));
+          const fingerprint = Buffer.from(data.slice(97, 101));
+          const xpub = hathorLib.wallet.xpubFromData(compressedPubkey, chainCode, fingerprint);
+          // First we clean what can still be there of a last wallet
+          wallet.cleanWallet();
+          wallet.startHardwareWallet(xpub);
+          hathorLib.wallet.markBackupAsDone();
+          this.props.history.push('/wallet/');
+        } else {
+          // Error
+          this.setState({ errorMessage: arg.error.message });
+        }
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    if (IPC_RENDERER) {
+      // Remove listeners
+      IPC_RENDERER.removeAllListeners("ledger:version");
+      IPC_RENDERER.removeAllListeners("ledger:publicKeyData");
+    }
+  }
+
+  /**
+   * Go to signin screen
```

Comment:
It's not signin anymore, it's warning screen

Reply:
Updated the docstring